### PR TITLE
Does a lot of molitz b changes part 2

### PR DESCRIPTION
**Pull request by ThatFiveGuys**
## About the PR 

This pr does a few things, first it reduces the amount of oxygen released from molitz b when its releasing agent b from 20 mols to 15 mols. This pr also gives raw molitz unique descriptions that gives hints about how much gas remains in the crystal. Finally for crystals, when they experience an explosion and the iterations left on the crystal is zero they make a unique sound, as an indicator something went right. Oh also these sounds now ignore space just so you can properly hear em in a depressurized place. This also refactors a lot of this code for code cleanliness reasons. Oh also fixes a bug to where inflammable materials would be flammable and also makes mechanic stuff inflammable so they dont get burned up while in the burn chamber.

Sound is from https://freesound.org/people/_stubb/sounds/389618/

## Why's this needed? 

Should make molitz b much more usable and give a lot more clues towards some aspects of it. 

## Changelog

```changelog
(u)Ikea
(*)Molitz crystals now come with unique examination text relating to how much gas is left inside of them
```

### CODE REVIEW

**Comment by Tarmunora** [comment]
this entire pull request is a bit of a mess tbh. Flammability bugfix is good, The rest... not really feeling it.

**Comment by ThatFiveGuys** [comment]
Currently theres no in game clues towards the fact that explosions allow you to get more usage out of molitz, a lot of the changes are done to accomplish that

**Comment by Tarmunora** [comment]
> Currently theres no in game clues towards the fact that explosions allow you to get more usage out of molitz, a lot of the changes are done to accomplish that. Also molitz sounds are cool and i want engineers to actually hear them hence the sound changes


Experimentation. There's no handholding wrt agent_b

**Comment by ThatFiveGuys** [comment]
> > Currently theres no in game clues towards the fact that explosions allow you to get more usage out of molitz, a lot of the changes are done to accomplish that. Also molitz sounds are cool and i want engineers to actually hear them hence the sound changes
> 
> Experimentation. There's no handholding wrt agent_b

The issue is theres literally nothing to point them towards experimenting that way

**Comment by Tarmunora** [comment]
> > > Currently theres no in game clues towards the fact that explosions allow you to get more usage out of molitz, a lot of the changes are done to accomplish that. Also molitz sounds are cool and i want engineers to actually hear them hence the sound changes
> > 
> > 
> > Experimentation. There's no handholding wrt agent_b
> 
> The issue is theres literally nothing to point them towards experimenting that way, the description text of “theres still small gas pockets” edges them towards the fact that maybe theres a way to get more bang for your buck but it doesnt tell you how. I think being able to figure stuff out in game without having to code dive or someone telling you is good

There's an entire department on most maps dedicated to A.) messsing with combustion and gasses and B.) using those gasses and combustion to blow shit up.

**Comment by ThatFiveGuys** [comment]
Has anyone ever actually found out that the explosion interaction is a thing on there own? Theres no train of logic there. If you want compensation for the additional clues i can add back molitz b being pushed by gas

**Comment by Azrun** [comment]
I think having the auditory cue ignore space is odd.  I understand you want to telegraph they did something correctly but that seems so different compared how most interactions work.
I think a line or two regarding the expectation being both temperature and explosive testing in the RD briefcase note could convey that.

**Comment by ThatFiveGuys** [comment]
The main reason for the sound ignoring space is just because engineers work in a depressurized enviroment meaning its basically impossible to hear them. I think the sounds are cool and i want them to be good background noise for when a hellburn is ongoing, not something that people never hear.

Adding a line about how explosions to briefcase would work, my main issue is you cant really hint it without giving it away or it not really making sense in universe and i think the description text hints at the possibility but doesnt give it away better

File: code/modules/materials/Mat_MaterialProcs.dm [hunk 1]
```diff
@@ -378,46 +376,57 @@ triggerOnEntered(var/atom/owner, var/atom/entering)
...
-			payload.temperature = T0C // Greatly reduce temperature to simulate an endothermic reaction
-			// Itr: .18 Agent B, 20 oxy, 1.3 minutes per iteration, realisticly around 7-8 minutes per crystal.
+			payload.temperature = T0C
 
-			animate_flash_color_fill_inherit(location,"#ff0000",4, 2 SECONDS)
-			playsound(location, 'sound/effects/leakagentb.ogg', 50, 1, 8)
-			if(!particleMaster.CheckSystemExists(/datum/particleSystem/sparklesagentb, location))
-				particleMaster.SpawnSystem(new /datum/particleSystem/sparklesagentb(location))
+			animate_flash_color_fill_inherit(owner,"#ff0000",4, 2 SECONDS)
+			playsound(owner, 'sound/effects/leakagentb.ogg', 50, 1, 8, flags = SOUND_IGNORE_SPACE)
```
Comment (virvatuli-gh) [inline code/modules/materials/Mat_MaterialProcs.dm:382]:
```suggestion
			playsound(owner, 'sound/effects/leakagentb.ogg', 50, 1, 8)
```

**Review by virvatuli-gh — CHANGES_REQUESTED** [review]
The ignore space flag is for things that you'd hear because they're in your suit like the geiger counter and similar, or notifications like the mentorhelp sting that should not be occluded by the reverb system. There may be other things that have it applied but I'd rather not see any more than necessary as they lead to sound amplification bugs when being spammed, see the reason why sleeper agent numbers were so dreadfully ear-destroying before I fixed them.

**Comment by Cherman0** [comment]
I like this conceptually, as currently molitz B is very underutilized on account of it being very unintuitive and convoluted to use and most of what the research note says being outright lies. Maybe its because I only play RP, but the most I've ever seen other people use molitz B is tossing it in the burn chamber and it doing nothing of note because that isnt really how it is effectively used (despite the research note suggesting that). 

On a somewhat related note, the concept of blowing it up never even occured to me. Why would it work? It doesnt make any sense logically and nothing suggests it or demonstrates it in some other context. I suppose I've also never tried deep frying it or tossing it in a portal or any number of other things which have no logical reason to do anything productive.

File: code/modules/materials/Mat_MaterialProcs.dm [hunk 2]
```diff
@@ -378,46 +376,57 @@ triggerOnEntered(var/atom/owner, var/atom/entering)
...
+			if(molitz.iterations == 0)
+				playsound(owner, 'sound/effects/molitzcrumble.ogg', 50, 1, 5, flags = SOUND_IGNORE_SPACE)
 			var/datum/gas_mixture/payload = new /datum/gas_mixture
 			payload.oxygen = 50
 			payload.temperature = T20C
 			target.assume_air(payload)
-			maxexplode -= 1
-			if(owner)
-				owner.setProperty("resonance", 1)
+			molitz.iterations = 2
```
Comment (Azrun) [inline code/modules/materials/Mat_MaterialProcs.dm:424]: Why 2?  Should it be += 2?

Seems like exploding it before using it is a weird trap or am I missing something?

File: code/obj/item/material.dm [hunk 3]
```diff
@@ -201,6 +201,25 @@
...
+		else if(molitz.iterations == 0 && molitz.unexploded == 0)
+			. += " All the gas in the crystal is completely gone."
+		else if(molitz.iterations >= 3)
+			. += " It looks like the crystal has plenty of gas trapped inside of it."
+		else if(molitz.iterations == 2)
+			. += " The crystal's large gas pockets look smaller then usual"
+		else if(molitz.iterations == 1)
+			. += " All of the big pockets of gas are nearly gone, however there are still plenty of small pockets of gas that contain more gas."
+		else if(molitz.iterations == 0)
+			. += " All the big pockets of gas are gone, yet the small pockets of gas refuse to vent."
```
Comment (Azrun) [inline code/obj/item/material.dm:221]: This seems oddly specific for a process I think is supposed to be discovered.  Also it makes a lot of assumptions of the actions you are performing on the item itself.  And what do gas pockets look like?  And what does it mean to be loose?

File: code/obj/item/material.dm [hunk 4]
```diff
@@ -213,6 +232,25 @@
...
```
Comment (Azrun) [inline code/obj/item/material.dm:252]: See previous.

**Review by Azrun — COMMENTED** [review]
I find this implementation of the iterations a lot easier to understand.  Not a fan of the specificity of the description text and assumption of the verbs the player is using. 

I do agree that visual indications make the most sense.

File: code/modules/materials/Mat_MaterialProcs.dm (hunk 2 above)
Comment (ThatFiveGuys) [reply code/modules/materials/Mat_MaterialProcs.dm:424]: The basic idea is that youre releasing all the big pockets of gas but youre fracturing the crystal enough to where the small pockets can leak. Basic idea is they use the sound + the examination text to notice oh i really messed it up and then learn from there

File: code/modules/materials/Mat_MaterialProcs.dm (hunk 1 above)
Comment (ThatFiveGuys) [reply code/modules/materials/Mat_MaterialProcs.dm]: Done!

File: code/modules/mechanics/MechanicMadness.dm [hunk 5]
```diff
@@ -347,6 +347,7 @@
 	plane = PLANE_NOSHADOW_BELOW
 	w_class = W_CLASS_TINY
 	level = 2
+	burn_possible = FALSE
```
[the remark this answers is not in the document]
Comment (ThatFiveGuys) [reply code/modules/mechanics/MechanicMadness.dm]: Forgot to remove those after making #11271, removed them from code now

File: code/obj/item.dm [hunk 6]
```diff
@@ -260,7 +260,7 @@
 		..()
 		tooltip_rebuild = 1
 		if (istype(src.material))
-			burn_possible = src.material.getProperty("flammable") >= 1 ? TRUE : FALSE
+			burn_possible = src.material.getProperty("flammable") > 1 ? TRUE : FALSE
```
[the remark this answers is not in the document]
Comment (ThatFiveGuys) [reply code/obj/item.dm]: Ditto

File: code/obj/item/material.dm (hunk 3 above)
Comment (ThatFiveGuys) [reply code/obj/item/material.dm]: Should be a lot better now

**Review by Azrun — APPROVED** [review]

**Comment by Azrun** [comment]
Per some discussion `sound/effects/molitzcrumble.ogg` be improved to feel more reactive and unique (more Sci-Fi), not required as part of this PR.

**Pull request merged**